orttraining/orttraining/core/graph/mixed_precision_transformer.cc

```diff
@@ -155,6 +155,17 @@ static Status CastNodeArg(onnxruntime::Graph& graph,
     return Status::OK();
   }
 
+  // If the producer is a Cast(to:float) node, and all consumers are mixed precision inputs, we can change
+  // the "to" attribute to mixed precision type of this Cast node directly. Otherwise, we still need to add
+  // a new Cast(to:elem_type) node for mixed precision inputs, and keep those fp32 inputs consuming the old node.
+  if (elem_type != ONNX_NAMESPACE::TensorProto_DataType_FLOAT && fp32_inputs.empty()) {
+    Node* producer_node = graph.GetMutableProducerNode(arg->Name());
+    if (producer_node != nullptr && producer_node->OpType() == "Cast") {
+      producer_node->AddAttribute("to", ONNX_NAMESPACE::MakeAttribute("to", int64_t(elem_type)));
+      return Status::OK();
+    }
+  }
+
```

Comment:
We have a RemoveDuplicateCastTransformer which you can leverage. And you may need to polish that transformer.

Reply:
Updated.